### PR TITLE
Add max client disconnect docs

### PR DESCRIPTION
**Pull request by mikenomitch**

### CODE REVIEW

File: website/content/docs/job-specification/group.mdx [hunk 1]
```diff
@@ -255,6 +260,75 @@ group "second" {
...
+and the allocation will be marked as "lost". If the client had reconnected
+before twelve hours had passed, the allocations would gracefully reconnect
+without a restart.
+
+Max Client Disconnect is useful for edge deployments, or scenarios when
+operators want zero on-client downtime due to node connectivity issues. This
+setting cannot be used with [`stop_after_client_disconnect`].
+
+```hcl
+# client_conifg.hcl
```
Comment (tgross) [inline website/content/docs/job-specification/group.mdx:304]:
```suggestion
# client_config.hcl
```

File: website/content/docs/job-specification/group.mdx [hunk 2]
```diff
@@ -81,19 +81,24 @@ job "docs" {
...
+- `stop_after_client_disconnect` `(string: "")` - Specifies a duration after
+  which a Nomad client will stop allocations, if it cannot communicate with the
+  servers. By default, a client will not stop an allocation until explicitly
+  told to by a server. A client that fails to heartbeat to a server within the
+  [`heartbeat_grace`] window and any allocations running on it will be marked
+  "lost" and Nomad will schedule replacement allocations. However, these
+  replaced allocations will continue to run on the non-responsive client; an
+  operator may desire that these replaced allocations are also stopped in this
+  case — for example, allocations requiring exclusive access to an external
+  resource. When specified, the Nomad client will stop them after this duration.
```
Comment (tgross) [inline website/content/docs/job-specification/group.mdx:93]: This sentence has run away on us a bit and has a bunch of passive voice and third-person reference to the operator. Maybe break it up into:

```suggestion
  "lost" and Nomad will schedule replacement allocations. The replaced 
  allocations will normally continue to run on the non-responsive client. But
  you may want them to stop instead — for example, allocations requiring 
  exclusive access to an external resource. When specified, the Nomad client 
  will stop them after this duration.
```

**Review by tgross — APPROVED** [review]
LGTM with some minor touch-ups.

**Comment by tgross** [comment]
Does this close https://github.com/hashicorp/nomad/pull/12448 ?

File: website/content/docs/job-specification/group.mdx [hunk 3]
```diff
@@ -255,6 +260,75 @@ group "second" {
...
+By default, allocations running on a client that fails to heartbeat will be
+marked "lost". When a client reconnects, its allocations, which may still be
+healthy, will restart because they have been marked "lost". This can cause
+issues with stateful tasks or tasks with long restart times.
+
+Instead, an operator may desire that these allocations reconnect without a
+restart. When `max_client_disconnect` is specified, the Nomad server will mark
+clients that fail to heartbeat as "disconnected" rather than "down", and will
+mark allocations on a disconnected client as "unknown" rather than "lost". These
+allocations will continue to run on the disconnected client. Replacement
```
Comment (DerekStrickland) [inline website/content/docs/job-specification/group.mdx:278]:
```suggestion
allocations may continue to run on the disconnected client. Replacement
```

**Review by DerekStrickland — APPROVED** [review]
Awesome job!

**Comment by mikenomitch** [comment]
Committed all the suggestions. Thanks.

I think it closes https://github.com/hashicorp/nomad/pull/12448 unless yall had something else in mind. We have the Edge Learn guide coming soon for in depth stuff too fwiw.

**Comment by tgross** [comment]
Final preview link: https://nomad-3ug4o9ram-hashicorp.vercel.app/docs/job-specification/group

**Pull request merged**